crates/node/src/service/rpc/mod.rs

```diff
@@ -0,0 +1,115 @@
+use std::sync::Arc;
+
+use jsonrpsee::server::ServerHandle;
+use tokio::task::JoinSet;
+
+use mc_db::DatabaseService;
+use mc_rpc::{providers::AddTransactionProvider, rpc_api_external, rpc_api_internal, Starknet};
+use mp_utils::service::Service;
+
+use metrics::RpcMetrics;
+use server::{start_server, ServerConfig};
+
+use crate::cli::{RpcMethods, RpcParams};
+
+use self::server::rpc_api_build;
+
+mod metrics;
+mod middleware;
+mod server;
+
+pub struct RpcService {
+    server_config_user: Option<ServerConfig>,
+    server_config_admin: Option<ServerConfig>,
+    server_handle_user: Option<ServerHandle>,
+    server_handle_admin: Option<ServerHandle>,
+}
+impl RpcService {
+    pub fn new(
+        config: &RpcParams,
+        db: &DatabaseService,
+        add_txs_method_provider: Arc<dyn AddTransactionProvider>,
+    ) -> anyhow::Result<Self> {
+        if config.rpc_disabled {
+            return Ok(Self {
+                server_config_user: None,
+                server_config_admin: None,
+                server_handle_user: None,
+                server_handle_admin: None,
+            });
+        }
+
+        let (rpcs, node_operator) = match (config.rpc_methods, config.rpc_external) {
+            (RpcMethods::Safe, _) => (true, false),
+            (RpcMethods::Unsafe, _) => (true, true),
+            (RpcMethods::Auto, false) => (true, true),
+            (RpcMethods::Auto, true) => {
+                tracing::warn!(
+                    "Option `--rpc-external` will hide node operator endpoints. To enable them, please pass \
+                     `--rpc-methods unsafe`."
+                );
+                (true, false)
+            }
+        };
```

Comment:
ensure this logic is explicitly reflected in the CLI

Reply:
I have updated the CLI docs to more accuratly reflect how this works :+1: